src/main/java/seedu/address/logic/commands/task/AssignTaskCommand.java

```diff
@@ -0,0 +1,129 @@
+            + "Example: " + COMMAND_WORD + " 1 " + "1";
+
+    public static final String MESSAGE_DUPLICATE_TASK = "Task '%s' is already assigned to %s";
+    private final Index personIndex;
+    private final Set<Index> taskIndexes;
+
+    /**
+     * Constructs a {@code TaskCommand} to assign an existing task to a person.
+     *
+     * @param perosnIndex The index of the person in the person list.
+     * @param taskIndexes The indexes of the tasks in the task list.
+     */
+    public AssignTaskCommand(Index perosnIndex, Set<Index> taskIndexes) {
+        requireNonNull(perosnIndex);
+        requireNonNull(taskIndexes);
+        this.personIndex = perosnIndex;
+        this.taskIndexes = taskIndexes;
+    }
+
+    /**
+     * Generates a command execution success message showing the added tasks and the person.
+     *
+     * @param personToEdit The person to whom the tasks were added.
+     * @return A success message indicating the tasks that were added and the name of the person.
+     */
+    private String generateSuccessMessage(Person personToEdit, Set<Task> taskToAdd) {
+        String addedTasks = taskToAdd.stream()
+                .map(task -> task.toString().replaceAll("[\\[\\]]", ""))
+                .collect(Collectors.joining(", "));
+        return String.format(MESSAGE_ADD_TASK_SUCCESS, addedTasks, personToEdit.getName().toString());
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Person> lastShownPersonList = model.getFilteredPersonList();
+        List<Task> lastShownTaskList = model.getFilteredTaskList();
+
+
+        if (personIndex.getZeroBased() >= lastShownPersonList.size()) {
+            throw new CommandException(MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownPersonList.get(personIndex.getZeroBased());
+
+        boolean isVendor = false;
+        for (Vendor vendor : model.getFilteredVendorList()) {
+            if (personToEdit.getName().equals(vendor.getName())) {
+                isVendor = true;
+                break;
+            }
+        }
+        if (!isVendor) {
+            throw new CommandException(MESSAGE_ONLY_VENDOR_CAN_BE_ASSIGNED_TASK);
+        }
+
+        Set<Task> tasksToAdd = new HashSet<>();
+        Set<Task> updatedTasks = new HashSet<>(personToEdit.getTasks());
+        for (Index taskIndex : taskIndexes) {
+            if (taskIndex.getZeroBased() >= lastShownTaskList.size()) {
+                throw new CommandException(MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+            }
+            Task newTask = lastShownTaskList.get(taskIndex.getZeroBased());
+            if (updatedTasks.contains(newTask)) {
+                throw new CommandException(String.format(MESSAGE_DUPLICATE_TASK,
+                        newTask.toString(), personToEdit.getName().toString()));
+            }
+            tasksToAdd.add(newTask);
+            updatedTasks.add(newTask);
```

Comment:
If you simply add the new task to the updatedTasks, it does not change the existing task. Would this result in duplicated tasks for a person?

Reply:
Yeap, there's a check before this for duplicate tasks and an exception will be thrown before the task can be added